lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/api/VortexThreadPool.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.vortex.api;
+
+import org.apache.reef.vortex.driver.tasklet.Tasklet;
+import org.apache.reef.vortex.driver.VortexManager;
+
+import javax.inject.Inject;
+import java.io.Serializable;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Distributed thread pool.
+ */
+public class VortexThreadPool {
+  private final AtomicInteger taskletId = new AtomicInteger();
+  private final VortexManager vortexManager;
+
+  @Inject
+  public VortexThreadPool(final VortexManager vortexManager) {
+    this.vortexManager = vortexManager;
+  }
+
+  public <TInput extends Serializable, TOutput extends Serializable> VortexFuture<TOutput>
+      submit(final VortexFunction<TInput, TOutput> function, TInput input) {
+    // todo: Support duplicate tasklets / tasklet priorities.
+    final VortexFuture<TOutput> vortexFuture = new VortexFuture<>();
+    vortexManager.enqueueNewTasklet(new Tasklet<>(taskletId.getAndIncrement(), function, input, vortexFuture));
```

Comment:
This interface's a bit strange.
`VortexManager` receives `Tasklet`s. You create a `VortexFuture` instance and pass it to the `Tasklet` instance. There's no return value from `enqueueNewTasklet`. 

I expected that `enqueueNewTasklet` returns a `VortexFuture` object which is created internally in the `enqueueNewTasklet` method.